client/packages/inventory/src/Stocktake/DetailView/Toolbar.tsx

```diff
@@ -12,26 +12,40 @@ import {
   DatePickerInput,
   Formatter,
   SearchBar,
-  useIsGrouped,
   Box,
   Switch,
   DateUtils,
   Alert,
+  useIsGrouped,
+  useUrlQuery,
 } from '@openmsupply-client/common';
 import { useStocktake } from '../api';
 
-export const Toolbar: FC = () => {
-  const t = useTranslation('inventory');
+export const Toolbar = () => {
+  const { isGrouped, toggleIsGrouped } = useIsGrouped('stocktake');
+  const [localIsGrouped, setLocalIsGrouped] = React.useState(isGrouped);
```

Comment:
this is the change to separate UI update with the wider state change, that impacts the table and slows thigns down.